### ISSUE TITLE
confusion, avoid duplication of subscription

### ISSUE THREAD

**Issue by kstan79** [hook0/hook0#12]
Hi, I'm trying to perform api create subscription, during development i notice that it created a lot of duplication in the subscription table. I'm recommend your let api developer manage the subscription id themself instead of your generate randomUUID at server side. That will greatly improve the development flow in the sense of:

1. developer able to link subscription id at their end, without need to go back their application update hook0 subscription id. that create process of match 2 system uuid
2. hook0 can block developer submit redundant uuid, that very help to prevent system bugs and submit redundant subscription

for backward compatibility, you can design the field as optional, if developer no specific uuid at new subscription it will create new one as usual

**Comment by dsferruzza** [comment]
Hi!

Thanks for you feedback!
We are currently in the process of stabilizing and delivering a big feature, but we will to try to look into this afterwards.
One solution to this issue may be for us to add PUT endpoints in addition to the current POST ones 🤔 

In the mid-time, you could store some kind of internal ID in the `metadata` field of your subscriptions, and use [this endpoint](https://documentation.hook0.com/reference/subscriptionslist) to list them and determine if you need to create one or not.

Have a nice day!

**Comment by kstan79** [comment]
yes I will have control mechanism of it. It is just a feed back to improve hook0 in developer perspective. you may close this issue if you acknowledge it, or leave it open for reference.

thanks of your support!
